src/core/int128.d

```diff
@@ -453,36 +453,135 @@ pure
 Cent udivmod(Cent c1, Cent c2, out Cent modulus)
 {
     //printf("udiv c1(%llx,%llx) c2(%llx,%llx)\n", c1.lo, c1.hi, c2.lo, c2.hi);
+    // Based on "Unsigned Doubleword Division" in Hacker's Delight
+    import core.bitop;
+
+    // Divides a 128-bit dividend by a 64-bit divisor.
+    // The result must fit in 64 bits.
+    static U udivmod128_64(Cent c1, U c2, out U modulus)
+    {
+        // We work in base 2^^32
+        enum base = 1UL << 32;
+        enum divmask = (1UL << (Ubits / 2)) - 1;
+        enum divshift = Ubits / 2;
+
+        // Check for overflow and divide by 0
+        if (c1.hi >= c2)
+        {
+            modulus = 0UL;
```

Comment:
Please add tests

Reply:
This is really an `assert(0)` case, as I can't see any scenario where this would happen.
1. In the 128/64 case, the value of `c1` is adjusted so that `c1.hi = c1.hi % c2`.
2. In the 128/128 case, `c2` is adjusted so that the MSB is set to 1 `c2 <<= clz(c2)`, and `c1` is adjusted so that the MSB is set to 0 `c1 >>= 1`.

However I don't want to add an `assert(0)` here.

Reply:
Ah right, since `modulus = 0` and `~0` is defined behaviour for poisoned calls, makes sense to leave that way and this seems unreachable anyway, I confirmed with a guided fuzzer.